### PR TITLE
exchange: expose GetCredentials() and split GetAuthenticatedAPISupport()

### PR DESCRIPTION
**Pull request by shazbert**
Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/interfaces.go [hunk 1]
```diff
@@ -125,3 +118,19 @@ type AccountManagement interface {
 	FetchAccountInfo(ctx context.Context, a asset.Item) (account.Holdings, error)
 	HasAssetTypeAccountSegregation() bool
 }
+
+// FunctionalityManager defines functionality for retrieving exchange
+// support/enabled features
+type FunctionalityManager interface {
```
Comment (gloriousCode) [inline exchanges/interfaces.go:124]: This doesn't manage anything it just checks if features are enabled or disabled

File: exchanges/interfaces.go [hunk 2]
```diff
@@ -95,6 +84,10 @@ type IBotExchange interface {
 	UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error
 
 	AccountManagement
+	GetCredentials(ctx context.Context) (*Credentials, error)
+	ValidateCredentials(ctx context.Context, a asset.Item) error
```
Comment (gloriousCode) [inline exchanges/interfaces.go:88]: Just an observation: I'm not opposed to this addition as it makes it easier, but these were already exposed using GetBase()

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Cool, just one thing

File: exchanges/interfaces.go (hunk 2 above)
Comment (shazbert) [reply exchanges/interfaces.go:88]: Yeah, that is very true, in future PR I will be updating account holdings to take in Credentials so we can segregate holdings to keys. Plus have DEX key support. I'm using this externally as well, so I thought, might as well.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Once you fix up testing, this should be good
```
2022-05-16T06:27:42.2611030Z panic: runtime error: invalid memory address or nil pointer dereference
2022-05-16T06:27:42.2611220Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x1a8 pc=0x1a0a444]
2022-05-16T06:27:42.2611248Z 
2022-05-16T06:27:42.2611351Z goroutine 1125 [running]:
2022-05-16T06:27:42.2611816Z github.com/thrasher-corp/gocryptotrader/engine.(*fakeDepositExchange).IsRESTAuthenticationSupported(0xc0002d10b0)
2022-05-16T06:27:42.2611950Z 	<autogenerated>:1 +0x44
2022-05-16T06:27:42.2612416Z github.com/thrasher-corp/gocryptotrader/engine.(*Engine).GetAllExchangeCryptocurrencyDepositAddresses.func1(0x0)
2022-05-16T06:27:42.2612747Z 	/home/runner/work/gocryptotrader/gocryptotrader/engine/helpers.go:701 +0x187
2022-05-16T06:27:42.2613356Z created by github.com/thrasher-corp/gocryptotrader/engine.(*Engine).GetAllExchangeCryptocurrencyDepositAddresses
2022-05-16T06:27:42.2613610Z 	/home/runner/work/gocryptotrader/gocryptotrader/engine/helpers.go:698 +0x106
```
https://github.com/thrasher-corp/gocryptotrader/runs/6447071529?check_suite_focus=true#step:7:3802

**Review by gloriousCode — APPROVED** [review]
tACK!
Auth still works:
`Binance: Exchange support: Enabled (Authenticated API support: Enabled - Verbose mode: Disabled).`
`Binance: Exchange support: Enabled (Authenticated API support: Disabled - Verbose mode: Disabled).`

**Review by thrasher- — APPROVED** [review]
ACK! Good stuff @shazbert

**Pull request merged**